### PR TITLE
chore: idempotent publish workflow

### LINKED ISSUE #1132: Fragile publish workflow
See error here https://github.com/winglang/wing/actions/runs/3896316568/jobs/6652941088#step:6:28

Occasionally, we fail to publish the vscode extension with this intermittent. At this point in the workflow, we already published NPM packages but have not yet created a release/tag. This unfortunately means that we cannot rerun the build with the same commit because it's won't increment the version (versions are based on last tag) and thus try (and fail) to publish packages.

The workflow should more resilient and/or auto-remediate the issue.

### PR DESCRIPTION
**Pull request by MarkMcCulloh**
- Create tag before actually publishing anything
  - This is needed to avoid a mix of things for a given version with the idempotent publishing

- vscode and npm will not attempt to publish if the version already exists

This is difficult to test, may have to iterate after this is merged.

Fixes #1132

*By submitting this pull request, I confirm that my contribution is made under the terms of the 
[Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -365,7 +365,23 @@ jobs:
         with:
           node-version: ${{ env.NODEJS_VERSION }}
 
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODEJS_VERSION }}
```
Comment (Chriscbr) [inline .github/workflows/build.yml:371]: Duplicate

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -365,7 +365,23 @@ jobs:
...
+
+      - name: Tag commit
+        uses: tvdias/github-tagger@v0.0.1
+        with:
+          repo-token: "${{ secrets.PROJEN_GITHUB_TOKEN }}"
+          tag: "v${{ needs.prepare.outputs.version }}"
```
Comment (Chriscbr) [inline .github/workflows/build.yml:377]: Is this step idempotent? If not can we make it idempotent?

Comment (MarkMcCulloh) [reply .github/workflows/build.yml:377]: It's not, I can see if I can get it to be though

Comment (MarkMcCulloh) [reply .github/workflows/build.yml]: done (hopefully)

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**